src/celeritas/ext/detail/CelerOpticalPhysics.cc

```diff
@@ -0,0 +1,337 @@
+#include <G4OpMieHG.hh>
+#include <G4OpRayleigh.hh>
+#include <G4OpWLS.hh>
+#include <G4ParticleDefinition.hh>
+#include <G4ProcessManager.hh>
+#include <G4Scintillation.hh>
+#include <G4Version.hh>
+#if G4VERSION_NUMBER >= 1070
+#    include <G4OpWLS2.hh>
+#    include <G4OpticalParameters.hh>
+#endif
+
+#include "corecel/Assert.hh"
+#include "corecel/io/Logger.hh"
+
+namespace celeritas
+{
+namespace detail
+{
+namespace
+{
+enum class OpticalProcessType
+{
+    cerenkov,
+    scintillation,
+    absorption,
+    rayleigh,
+    mie_hg,
+    boundary,
+    wavelength_shifting,
+    wavelength_shifting_2,
+    size_,
+};
+
+#if G4VERSION_NUMBER >= 1070
+G4OpticalProcessIndex
+optical_process_type_to_geant_index(OpticalProcessType value)
+{
+    switch (value)
+    {
+        case OpticalProcessType::cerenkov:
+            return kCerenkov;
+        case OpticalProcessType::scintillation:
+            return kScintillation;
+        case OpticalProcessType::absorption:
+            return kAbsorption;
+        case OpticalProcessType::rayleigh:
+            return kRayleigh;
+        case OpticalProcessType::mie_hg:
+            return kMieHG;
+        case OpticalProcessType::boundary:
+            return kBoundary;
+        case OpticalProcessType::wavelength_shifting:
+            return kWLS;
+        case OpticalProcessType::wavelength_shifting_2:
+            return kWLS2;
+        default:
+            return kNoProcess;
+    }
+}
+
+G4String optical_process_type_to_geant_name(OpticalProcessType value)
+{
+    return G4OpticalProcessName(optical_process_type_to_geant_index(value));
+}
+#endif
+
+//---------------------------------------------------------------------------//
+/*!
+ * Return true if a given process is active
+ *
+ * Use `G4OpticalParameters` when available, otherwise use hardcoded
+ * checks.
+ */
+bool process_is_active(
+    OpticalProcessType process,
+    [[maybe_unused]] CelerOpticalPhysics::Options const& options)
+{
+#if G4VERSION_NUMBER >= 1070
+    auto* params = G4OpticalParameters::Instance();
+    CELER_ASSERT(params);
+    return params->GetProcessActivation(
+        optical_process_type_to_geant_name(process));
+#else
+    switch (process)
+    {
+        case OpticalProcessType::cerenkov:
+            return bool(options.cerenkov);
+        case OpticalProcessType::scintillation:
+            return bool(options.scintillation);
+        case OpticalProcessType::absorption:
+            return options.absorption;
+        case OpticalProcessType::rayleigh:
+            return options.rayleigh_scattering;
+        case OpticalProcessType::mie_hg:
+            return options.mie_scattering;
+        case OpticalProcessType::boundary:
+            return bool(options.boundary);
+        case OpticalProcessType::wavelength_shifting:
+            return options.wavelength_shifting != WLSTimeProfileSelection::none;
+        case OpticalProcessType::wavelength_shifting_2:
+            // Not supported pre 10.7
```

Comment:
I know we're not worrying about any version < 10.7, but should we throw a `CELER_ASSERT_UNREACHABLE();` here?